dev_guide/builds.adoc

```diff
@@ -519,9 +531,42 @@ For example, defining a custom HTTP proxy to be used during build:
 ----
 ====
 
+[[source-code]]
+
+== Git Repository Source Options
```

Comment:
Nesting those into new top-level sections was my first approach as we discussed on IRC the other day, and I have nothing against going into that direction eventually.

To facilitate review though, I opted for just adding sections for now until we finalize the content, and later move things around. Doing all at once makes the diff pretty unreadable.

As I noticed on IRC, adding the new top-level sections makes leaf subsections to be in the lowest possible nesting level, 5.
We may investigate whether moving (sub)content into subpages (linked from builds.html) would work well to make content more browseable and discoverable.